apps/prairielearn/src/pages/instructorGradingJob/instructorGradingJob.ts

```diff
@@ -33,13 +33,12 @@ router.get(
       throw new error.HttpStatusError(404, 'Job not found');
     }
 
-    // If the grading job is associated with an assessment instance (through a
-    // submission, a variant, and an instance question), then we need to check
-    // if the effective user is authorized to view this assessment instance.
-    //
-    // The way we implement this check right now with authz_assessment_instance
-    // is overkill, yes, but is easy and robust (we hope).
-    if (gradingJobRow.aai && !gradingJobRow.aai.authorized) {
+    // 'instanceAuthorized' may be null on course pages.
+    const instanceAuthorized = res.locals.authz_data.has_course_instance_permission_view ?? false;
+    const courseAuthorized = res.locals.authz_data.has_course_permission_preview;
+
+    // course_role >= 'Previewer' OR course_instance_role >= 'Student Data Viewer'
+    if (!instanceAuthorized && !courseAuthorized) {
```

Comment:
I don't believe this is a completely faithful rewrite of the checks from the sprocs. Specifically, it seems like this grants course Previewers access to student data.

I think the missing piece is this:

https://github.com/PrairieLearn/PrairieLearn/blob/c34f0ba7e98d6e0112efd6db782bf154c9b6b3c9/apps/prairielearn/src/sprocs/authz_assessment_instance.sql#L110-L116

There's a huge comment above that that explains why it's necessary, I'd encourage you to read through it.

Reply:
I see. What I am getting from this is that if the user is not the owner, then they need also need course instance student viewer permissions. We need to check this for both group assessment instances and non-group assessment instances. 

A suggestion would be to return both `assessment_instances` and the list of group users associated with the instance in `instructorGradingJob.sql`. Then, I have enough information to evaluate this extra check for assessment instances in typescript.

Reply:
I did this in af31821d8cd7e4c11c52a17668e5fe8d0d6f0f52 with whole row returns. I can pull off smaller pieces if wanted.